src/plot_api/subroutines.js

```diff
@@ -79,10 +79,8 @@ exports.lsInner = function(gd) {
             return;
         }
 
-        var xa = Plotly.Axes.getFromId(gd, subplot, 'x');
-        var ya = Plotly.Axes.getFromId(gd, subplot, 'y');
-        var xDomain = xa.domain;
-        var yDomain = ya.domain;
+        var xDomain = plotinfo.xaxis.domain;
```

Comment:
Amazing. Note that references are updated in `plotinfo` in [`linkSubplots`](https://github.com/plotly/plotly.js/blob/edf52ca901923959023feeb351ab7f3d4e82faab/src/plots/plots.js#L633) called during supplyDefaults.